pkg/auth/authproviders/identity.go

```diff
@@ -32,6 +32,13 @@ func CreateRoleBasedIdentity(ctx context.Context, provider Provider, authResp *A
 		Attributes: authResp.Claims.Attributes,
 	}
 
+	// The attribute checker is not required for every provider, only check if set.
```

Comment:
I'm still not sure this is the right place to call `AttributeChecker`. I understand that `CreateRoleBasedIdentity` is the single point of transformation between ` *AuthResponse` and `*v1.AuthStatus`, but method name and body do not really correlate with the fact that this check is here.

Reply:
Hm, I get your point, similar to the `authResp.Claims == nil` check, its currently doing more than expected.
The issue I see with potentially splitting this into two calls would be if clients, like the `authprovider` service, which currently calls `CreateRoleBasedIdentity` would suddenly also make sure they verify it beforehand. I don't think we should move that responsibility to them.

I don't necessarily have a _better_ name for the function itself, not something that would be of the likes of `VerifyAndCreate..` which is _meh_ 😞

Reply:
I feel like we are jumping a bit around the topics, on one hand I agree that doing it within `CreateRoleBasedIdentity` is not desirable, we should clarify this within this PR what we ultimately decide to do here.

Regarding the additional checks within `ExchangeToken` that may or may not be moved to `registry_httphandler`, I'd vote to address them later on. The PR is already quite big and it would increase scope & complexity.

Coming back to `CreateRoleBasedIdentity` - what do you think, should we keep as is, with the comment as a hint that _more_ is done ? Personally, I'm fine with the additional comment.

Reply:
FWIW, I'm not quite happy with expanding `CreateRoleBasedIdentity()` either but I don't have a better suggestion. Specifically, it's unfortunate that we deny login attempt _after_ the auth provider has given green light. Ideally, we would have a generic `ProcessHTTPRequest()` implementation that calls specific version first and then does the required attribute check. Not sure what's the best way to express it in Golang.

Reply:
@ivan-degtiarenko I see, now I understood better what you mean. I could agree with this for the time being.  I've moved the check within `providersHTTPHandler`.